walk-in-interview/src/main/java/com/google/job/servlets/ChangeJobStatusServlet.java

```diff
@@ -0,0 +1,101 @@
+package com.google.job.servlets;
+
+import com.google.job.data.JobStatus;
+import com.google.job.data.JobsDatabase;
+import com.google.utils.ServletUtils;
+
+import javax.servlet.ServletException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+
+/** Servlet that handles changing status of the existing job posts. */
+@WebServlet("/jobs/change-job-status")
+public final class ChangeJobStatusServlet extends HttpServlet {
+    private static final String PATCH_METHOD_TYPE = "PATCH";
+    private static final String JOB_ID_FIELD = "jobId";
+    private static final String JOB_STATUS_FIELD = "jobStatus";
+    private static final long TIMEOUT = 5;
+
+    private JobsDatabase jobsDatabase;
+
+    @Override
+    public void init() {
+        this.jobsDatabase = new JobsDatabase();
+    }
+
+    @Override
+    public void service(HttpServletRequest request, HttpServletResponse response) throws ServletException, IOException {
+        // Explicitly routes PATCH requests to a doPatch method since by default HttpServlet doesn't do it for us.
+        if (request.getMethod().equalsIgnoreCase(PATCH_METHOD_TYPE)){
+            doPatch(request, response);
+        } else {
+            super.service(request, response);
+        }
+    }
+
+    /** Handles the PATCH request from client. */
+    public void doPatch(HttpServletRequest request, HttpServletResponse response) {
+        try {
+            // Gets the target job post id
+            String jobId = ServletUtils.getStringParameter(request, JOB_ID_FIELD, /* defaultValue= */ "");
+
+            // Gets the target status
+            JobStatus jobStatus = getTargetStatus(request);
```

Comment:
Maybe we should not let frontend control this status by directly modifying the field? Maybe frontend indicate the desire "action" instead? 
E.g Instead of frontend setting the new status as deleted, we allow frontend to send a DELETE http request, on server, we handle in doDelete(). Inside doDelete() we mark status as DELETED or whatever status we desire as it's internal logic of the backend? Maybe in the future, we can even change it to "INACTIVE" and it won't affect frontend?

Reply:
As discussed offline, change to a new end point "/jobs/delete"